codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -652,50 +652,57 @@ let actionReducer = (state: t, action: action) => {
       ],
     )
   | StartIndexingAfterPreRegister =>
```

Comment:
Looks like a hot fix. Didn't you manage to find why the StartIndexingAfterPreRegister is dispatched multiple times?

Reply:
To be honest, I haven't been able to recreate the exact case where this can occur other than observing it on 2 different indexers.

Running through all the cases I can't see how this action ends up being dispatched twice. I thought it might be a race condition between when a block range is returning and when a batch is finished processing. Wherever this case happens there needs to be an idempotency check at some level. I thought it's safest to do it inside the action that actually runs the restart of the indexer.